packages/block-editor/src/components/block-list/block.js

```diff
@@ -459,6 +459,7 @@ function BlockListBlock( {
 		<BlockEdit
 			name={ name }
 			isSelected={ isSelected }
+			isParentOfSelectedBlock={ isParentOfSelectedBlock }
```

Comment:
Any prop that we pass to a block's `edit` component will become a part of the official API for building blocks. This means that we'll have to support it forever and ever. I don't think that `isParentOfSelectedBlock` has enough use cases to be an official API.

Could we do this entirely with CSS? I note that `isParentOfSelectedBlock` is used to add the `.has-child-selected` class to the block's container. Could we hide `.block-editor-inner-blocks` when the block container does **not** have the `.has-child-selected` class?

If that doesn't work, then I'd prefer that we define `isParentOfSelectedBlock` in the Navigation block's `edit` component using `select( 'core/block-editor' ).hasSelectedInnerBlock( clientId, true )`. It means that we'll be repeating ourselves a little but I think that's better than committing to an API that might not be broadly useful.

Reply:
`.has-child-selected` doesn't work for us because you would need to see the child to first select it and only then the container will get the class. Because of various containers added by the framework :first-child, :first-of-type and :nth-child don't work in this situation.

There is an argument to have the API support even more of this kinds of flags, as a service to block implementers who might not be aware of how to find them in the various stores for selection (isSelected, isParentOfSelectedBlock, isChildOfSelectedBlock, isFirstInnerBlock, isLastInnerBlock etc). It is a burden to support but it is also a nicety and they won't go away as long as we have trees of blocks as a structure, you'll always need to know these sort of flags (like in CSS!)

I will try your suggestion with selecting from the block editor store for now and be back

Reply:
Check it out now :)

Reply:
You may well be right! Since this is such a highly visible API for block developers, though, I'd prefer to play it safe and wait to see if there are _multiple_ use-cases for such an API before adding it.

Reply:
(I'm reminded of this post from a while back: https://dev.to/wuz/stop-trying-to-be-so-dry-instead-write-everything-twice-wet-5g33)